weblate/api/views.py

```diff
@@ -607,8 +616,16 @@ def update(self, request: Request, *args, **kwargs):
 
     def create(self, request: Request, *args, **kwargs):
         """Create a new group."""
-        self.perm_check(request)
-        return super().create(request, *args, **kwargs)
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        self.perm_check(
+            request, project=serializer.validated_data.get("defining_project")
+        )
+
+        self.perform_create(serializer)
```

Comment:
Wouldn't it be simpler to move the permission check inside `perform_create` and avoid overriding `create`?

Reply:
Ah yeah makes sense.